hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -976,6 +989,12 @@ private void startJoin() {
                 final int count = members.size() - 1 + setJoins.size();
                 final List<Future> calls = new ArrayList<Future>(count);
                 for (MemberInfo member : setJoins) {
+                    if (node.isMaster()) {
+                        //Only master node should send clusterId and startTime
+                        long startTime = clusterClock.getClusterStartTime();
+                        calls.add(invokeClusterOperation(new FinalizeJoinOperation(memberInfos, postJoinOp, time,
```

Comment:
Yes, you are right ```startJoin()``` is called only by master node, I misunderstood the logic about time in this code, 
```
if (firstJoinRequest == 0) {
	firstJoinRequest = now;
}

if (setJoins.add(memberInfo)) {
	sendMasterAnswer(joinRequest.getAddress());
	if (now - firstJoinRequest < maxWaitMillisBeforeJoin) {
    	timeToStartJoin = now + waitMillisBeforeJoin;
	}
}
if (now > timeToStartJoin) {
	startJoin();
}
``` 
I removed unnecessary parts.